### PR TITLE
YLP-1943 Training material modal should open the document before enabling access to yourloops 

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

File: locales/en/yourloops.json [hunk 1]
```diff
@@ -266,6 +267,7 @@
   "notification-patient-invitation-by-team": "You're invited to share your diabetes data with <strong>{{careteam}}</strong>.",
   "notification-patient-invitation-wrong-code": "The team code does not match the one of {{careteam}}.",
   "open-team-menu": "Open team menu",
+  "open-training": "Open {{training}}",
```
Comment (fatb38) [inline locales/en/yourloops.json:270]: I don't figure out why adding a variable training ? Is the text subject to change ?

**Review by fatb38 — CHANGES_REQUESTED** [review]

File: packages/yourloops/pages/training/training.tsx [hunk 2]
```diff
@@ -83,6 +92,22 @@ function TrainingPage(): JSX.Element {
   const fromPath = historyHook.location.state?.from?.pathname
   const user = auth.user
   const classes = style()
+  const [trainingOpened, setTrainingOpened] = useState(false)
+  const [checked, setChecked] = useState(false)
+
+  const checkbox = (
```
[the remark this answers is not in the document]
Comment (yanngit) [reply packages/yourloops/pages/training/training.tsx:98]: done

File: locales/en/yourloops.json (hunk 1 above)
Comment (yanngit) [reply locales/en/yourloops.json:270]: you're right it's always training I think

File: packages/yourloops/pages/training/training.tsx [hunk 3]
```diff
@@ -124,22 +134,55 @@ function TrainingPage(): JSX.Element {
...
+                    target="_blank"
+                    rel="noreferrer"
+                  >
+                    <Button
+                      variant="contained"
+                      color="primary"
+                      disableElevation
+                      onClick={openTraining}
+                    >
+                      {t('open-training', { training })}
```
Comment (fatb38) [inline packages/yourloops/pages/training/training.tsx:182]: Could be removed now it's no more in translation
```suggestion
                      {t('open-training')}
```

File: locales/en/yourloops.json [hunk 4]
```diff
@@ -391,7 +392,8 @@
   "today": "Today",
   "tooltip-release-notes": "Release notes",
   "training": "Training",
-  "training-body": "New <trainingLink>{{training}}</trainingLink> available, please read what's new before continuing on yourloops. You can also read it later, by clicking in the {{training}} footer link.",
+  "training-body": "New {{training}} available, please read what's new before continuing on yourloops.",
```
Comment (fatb38) [inline locales/en/yourloops.json:395]: `{{training}}` variable could be removed too (also in the component)
```suggestion
  "training-body": "New training available, please read what's new before continuing on yourloops.",
```

File: packages/yourloops/pages/training/training.tsx [hunk 5]
```diff
@@ -96,12 +107,11 @@ function TrainingPage(): JSX.Element {
     })
   }
 
+  const openTraining = (): void => {
+    setTrainingOpened(true)
+  }
+
   const training = t('training').toLowerCase()
```
Comment (fatb38) [inline packages/yourloops/pages/training/training.tsx:114]: becomes useless no ?

File: packages/yourloops/pages/training/training.tsx [hunk 6]
```diff
@@ -124,22 +134,55 @@ function TrainingPage(): JSX.Element {
...
+                  id={'form-control-training'}
+                  control={
+                    <Checkbox
+                      id={'checkbox-training'}
+                      aria-label={t('training-checkbox')}
+                      className={classes.checkbox}
+                      checked={checked}
+                      onChange={() => {
+                        setChecked(true)
+                      }}
```
Comment (fatb38) [inline packages/yourloops/pages/training/training.tsx:148]: Could be on one line (better readability)
```suggestion
                      onChange={() => setChecked(true)}
```

**Review by fatb38 — CHANGES_REQUESTED** [review]

**Review by fatb38 — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,10 +2,11 @@
 Blip is the web front end for YourLoops system.
 It is based on Tidepool Blip 1.27.
 
-## 3.0.2-beta32 - 2022-11-21
+## 3.0.2-beta33 - 2022-11-25
 ### Added
 - YLP-1679 Allow caregivers to remove patients from their direct share
 - YLP-1850 Patient search should comply to INS standard
+- YLP-1943 Training material modal should open the document before enabling access to yourloops
```
Comment (franck-fourel) [inline CHANGELOG.md:9]:
```suggestion
- YLP-1943 Training material is acknowledged after Yourloops TM is opened and read (including the new checkbox message) and before any use of YLP
```

**Review by TimCanu — APPROVED** [review]

**Pull request merged**